test/testdata/infer/generics/apply_f.rb

```diff
@@ -50,9 +50,9 @@ def takes_any_int_string(y)
   end
   T.reveal_type(x) # error: `T.any(Integer, String)`
 
-  x = apply_f_int(y) do |x| # error: Could not find valid instantiation of type parameters for `Object#apply_f_int`
```

Comment:
This error being reported was actually a bug, and I simply didn't realize it at the time.

The lack of an error here and an inferred type of `Integer` is actually what we'd expect.